azure/iac/registry.rego

```diff
@@ -13,15 +20,23 @@ azure_issue ["adminUserEnabled"] {
 
 adminUserEnabled {
     lower(input.resources[_].type) == "microsoft.containerregistry/registries"
+    not azure_attribute_absence["adminUserEnabled"]
     not azure_issue["adminUserEnabled"]
 }
 
+adminUserEnabled = false {
+    azure_attribute_absence["adminUserEnabled"]
+}
+
 adminUserEnabled = false {
     azure_issue["adminUserEnabled"]
 }
 
+adminUserEnabled_miss_err = "Container Registry attribute adminUserEnabled missing in the resource!" {
+    azure_issue["adminUserEnabled"]
+}
 
-adminUserEnabled_err = "ENSURE THAT ADMIN USER IS DISABLED FOR CONTAINER REGISTRY" {
+adminUserEnabled_err = "Ensure that admin user is disabled for Container Registry" {
```

Comment:
if ```adminUserEnabled``` does not exist that means admin user is not enabled for the ACR.
so we don't need to check existence of that
just we should make sure the value of ```adminUserEnabled`` is set to false if ```adminUserEnabled``` exists

so I fixed the issue which you mentioned
@rezoan

Reply:
@shahinshirvani you need to check the existence of the property at first point if this does not exist calling rule `adminUserDisabled` should return true.
Please add this:
```

azure_attribute_absence ["adminUserDisabled"] {
    resource := input.resources[_]
    lower(resource.type) == "microsoft.containerregistry/registries"
    not resource.properties.adminUserEnabled
}

adminUserDisabled {
    azure_attribute_absence["adminUserDisabled"]
}
```

Reply:
@shahinshirvani 
I agree it works in this scenario but this is not my point. 
when `adminUserEnabled` does not exist, you are getting `true `because `not undefined` is always `true`.
If you call the `azure_issue["adminUserDisabled"]` separately with non existing `adminUserEnabled` it will give you an `undefiled ` output. this is my concern. because `resource.properties.adminUserEnabled != false` will evaluate as `undefined != false` -> output: `undefiled`.
We should not make any final decision based on undefined output. if you do, you will not able to tell whether the output came due to property missing or misconfigured value. 
**Try to make it more readable by separating the property existence checking and its value checking.** in this way we can provide more exact error message.

I hope you understand my point here. its not about what is the output, its all about how we are generating the output.

Reply:
@rezoan @shahinshirvani i will close this PR, @shahinshirvani  please make sure you are following best practices Rezoan is mentioning here

Reply:
@rezoan I got your point and fixed the issue